src/components/vm/consoles/vncBody.jsx

```diff
@@ -34,34 +34,30 @@ export const VncRow = ({ idPrefix, onValueChanged, dialogValues, validationError
 
     return (
         <>
-            <Grid hasGutter md={6}>
-                <GridItem span={6}>
-                    <FormGroup fieldId={`${idPrefix}-address`} label={_("Listening address")}>
-                        <TextInput id={`${idPrefix}-address`}
-                            value={dialogValues.vncAddress}
-                            type="text"
-                            placeholder={_("default")}
-                            onChange={(event) => onValueChanged('vncAddress', event.target.value)} />
-                    </FormGroup>
-                </GridItem>
-                <GridItem span={6}>
-                    <FormGroup fieldId={`${idPrefix}-port`} label={_("Listening port")}>
-                        <TextInput id={`${idPrefix}-port`}
-                            value={dialogValues.vncPort}
-                            type="text"
-                            placeholder={_("automatic")}
-                            validated={validationErrors.vncPort ? "error" : null}
-                            onChange={(event) => onValueChanged('vncPort', event.target.value)} />
-                        { validationErrors.vncPort &&
-                            <FormHelperText>
-                                <HelperText>
-                                    <HelperTextItem variant='error'>{validationErrors.vncPort}</HelperTextItem>
-                                </HelperText>
-                            </FormHelperText>
-                        }
-                    </FormGroup>
-                </GridItem>
-            </Grid>
+            <FormGroup
+                fieldId={`${idPrefix}-portmode`}
+                label={_("Listening")} isInline hasNoPaddingTop isStack>
+                <Checkbox
+                    id={`${idPrefix}-autoport`}
+                    label={_("Use a static port")}
```

Comment:
I find this rather "meh". It neither explains what the default is (i.e. dynamic port, not "no port/no server"), or why you'd want one. TBH I liked the initial simple "Port: [  ]" text input where empty was supported with an "automatic" placeholder. Fewer moving parts and easier to understand IMHO.

Reply:
PF says: "[Placeholder text](https://www.patternfly.org/components/forms/form/design-guidelines/#placeholder-text) is text inside a form field that provides an example of the required or recommended format for text input. For example, placeholder text for a date might be "DD/MM/YYYY"."

So we can't use the default value, if we follow PF. We would have to put a helper text under the field that says "Leave empty to assign a free port automatically when VM starts." or similar.

I'll do that.